SLL.java

```diff
@@ -16,6 +17,7 @@ public SLL() {
 		head = null;
 		size = 0;
 		comparator = null;
+		tail = null;

```

Comment:
The add, remove, and addInOrder methods need changing to manage the tail field

Reply:
Okay I'll try to implement that tomorrow